torax/_src/transport_model/combined.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2024 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The CombinedTransportModel class.
+
+A class for combining transport models.
+"""
+import chex
+from jax import numpy as jnp
+
+from torax._src import array_typing
+from torax._src import state
+from torax._src.config import runtime_params_slice
+from torax._src.geometry import geometry
+from torax._src.pedestal_model import pedestal_model as pedestal_model_lib
+from torax._src.transport_model import runtime_params as runtime_params_lib
+from torax._src.transport_model import transport_model
+
+
+@chex.dataclass(frozen=True)
+class DynamicRuntimeParams(runtime_params_lib.DynamicRuntimeParams):
+  transport_model_params: Sequence[runtime_params_lib.DynamicRuntimeParams]
+
+
+class CombinedTransportModel(transport_model.TransportModel):
+  """Combines coefficients from a list of transport models."""
+
+  def __init__(
+      self, transport_models: Sequence[transport_model.TransportModel]
+  ):
+    super().__init__()
+    self.transport_models = transport_models
+    self._frozen = True
+
+  def _call_implementation(
+      self,
+      dynamic_runtime_params_slice: runtime_params_slice.DynamicRuntimeParamsSlice,
+      geo: geometry.Geometry,
+      core_profiles: state.CoreProfiles,
+      pedestal_model_outputs: pedestal_model_lib.PedestalModelOutput,
+  ) -> state.CoreTransport:
+    assert isinstance(
+        dynamic_runtime_params_slice.transport, DynamicRuntimeParams
+    )
+
+    summed_transport_coeffs = state.CoreTransport.zeros(geo)
+
+    for component_model, component_params in zip(
+        self.transport_models,
+        dynamic_runtime_params_slice.transport.transport_model_params,
+    ):
+      # Each transport model expects dynamic_runtime_params_slice.transport to
+      # be its corresponding DynamicRuntimeParams
+      # TODO: This seems like a hacky way of getting round what is potentially
+      # a problem with the design [TB 22 May 2025]
+      component_dynamic_runtime_params_slice = dataclasses.replace(
+          dynamic_runtime_params_slice,
+          transport=component_params,
+      )
```

Comment:
@tamaranorman thoughts on this? Bit clunky, but I didn't know how far to unravel.

Example: `critical_gradient.CriticalGradientModel` expects `dynamic_runtime_params_slice.transport` to be of type `critical_gradient.DynamicRuntimeParams`.

https://github.com/google-deepmind/torax/blob/3c8c7764d31369651ed42edec8cd2f0d4608dca2/torax/_src/transport_model/critical_gradient.py#L79C1-L81C6